lib/applications.js

```diff
@@ -1,15 +1,26 @@
 'use strict';
 
+const semver = require('semver');
+
 const errors = require('./errors');
 
 // Generic application class. You are free to substitute it with whatever suits
 // your needs.
-//   name - application name
-//   api - application API
+//   name - application name that may contain version after '@'
+//          (e.g. app@1.0.0). Version in name is preferred over
+//          'version' parameter
+//   api - object that contains interfaces as its fields each of which
+//         contains functions by method names. Each method has the following
+//         signature (connection, <0 or more method arguments>, callback)
+//   version - application version of 'name' application
 //
 class Application {
-  constructor(name, api) {
-    this.name = name;
+  constructor(name, api, version) {
+    [this.name, this.version] = name.split('@');
```

Comment:
@belochub I don't find it a valid justification. Why provide two ways to do the same thing with one of them being subtly broken? Either the npm-ish format must work according to users' expectations, or the `{ name, version }` schema must be the only available contract.